### DISCUSSION TITLE
How to update a installed device or check if there is any new device with better compatibility?

### DISCUSSION THREAD

**Discussion by ceinmart** [make-all/tuya-local#1437] [Q&A] [answered]
Hi , 

I have a few devices, that work well, however, I notice someones aren't 100% compatible, not show all the resources they have. 
They are functional and in use on my Home Assistant. 

I want to check if have new devices on Tuya Local with better compatibility and update my installed devices with them. 
Really would like to do this without deleting and reinstalling these devices because HA has a very annoying behavior to not accepting the same ID of a device deleted... and then this messes with all my dashboards, scripts, and automation... 

Is there a way to do this?
Or is plausible to be a feature request?  Upgrade an installed device with a better device Yaml?

**Comment by make-all** [comment, marked as answer]
You cannot change the config without deleting and re-adding the device.  Different configs define different entities, so to simply change the config would create a mess in the entity registry within Home Assistant.

**Comment by ceinmart** [comment]
Hi Jason, 

Thank you for the explanation. 
It's possible to check for compatibility of new/updated devices implemented without remove my current device? 
or is it plausible to be a feature request? 
Let's say... into the configuration screen a button to check the compatibility with the current tuyalocal database of devices... 
Showing the % match like in the logs... 
That way at least we can know there is an advantage to remove/add and reorganize all again...

**Comment by make-all** [comment]
This seems like a good idea.  I started working on better displaying the options (showing what current entity values are for each config instead of just the filename), and this same feature could be available from the options dialog for informational purposes.
